in_toto/verifylib.go

```diff
@@ -0,0 +1,502 @@
+package in_toto
+
+import (
+  "fmt"
+  "time"
+  "strings"
+  "reflect"
+  osPath "path"
+  "path/filepath"
+)
+
+func RunInspections(layout Layout) (map[string]Metablock, error) {
+  inspectionMetadata := make(map[string]Metablock)
+
+  for _, inspection := range layout.Inspect {
+
+    linkMb, err := InTotoRun(inspection.Name, []string{"."}, []string{"."},
+        inspection.Run)
+    if err != nil {
+      return nil, err
+    }
+
+    retVal := linkMb.Signed.(Link).ByProducts["return-value"]
+    if retVal != 0 {
+      return nil, fmt.Errorf("Inspection command '%s' of inspection '%s'" +
+          " returned a non-zero value: %d", inspection.Run, inspection.Name,
+          retVal)
+    }
+
+    // Dump inspection link to cwd using the short link name format
+    linkName := fmt.Sprintf(LinkNameFormatShort, inspection.Name)
+    linkMb.Dump(linkName)
+
+    inspectionMetadata[inspection.Name] = linkMb
+  }
+  return inspectionMetadata, nil
+}
+
+func Subtract(a []string, b []string) []string {
```

Comment:
This is a set difference function in `O(n**2)`, right? If you expect large `a` and `b`, perhaps consider the linear-time version over here, https://siongui.github.io/2018/03/14/go-set-difference-of-two-arrays/ , or find a proper set library :-)

Reply:
Oh yes, I was considering using the more performant map *hack*, but opted for the other one for reasons of readability, and because I thought `O(n**2)` is *okay* for now. But you're right. I will at least add a comment.

Reply:
Saw your comment in 0d6b3c065c931996ba62eac43fe58f5fb7e50876 and liked it!